olp-cpp-sdk-core/tests/client/HRNTest.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <olp/core/client/HRN.h>
+
+#include <gtest/gtest.h>
+#include <future>
+
+using olp::client::HRN;
+
+namespace {
+
+TEST(HRNTest, InitializeWithStringInvalid) {
+  {
+    SCOPED_TRACE("Empty HRN");
+    HRN hrn("");
+    EXPECT_TRUE(hrn.IsNull());
+  }
+  {
+    SCOPED_TRACE("Invalid HRN 1");
+    HRN hrn("invalid_hrn");
+    EXPECT_TRUE(hrn.IsNull());
+  }
+  {
+    SCOPED_TRACE("Invalid HRN 2");
+    HRN hrn("hrn:invalid_hrn");
+    EXPECT_TRUE(hrn.IsNull());
+  }
+  {
+    SCOPED_TRACE("Catalog HRN Tests");
+    {
+      SCOPED_TRACE("Valid Catalog HRN");
+      HRN hrn("hrn:here:data::test:hereos-internal-test-v2");
+      EXPECT_FALSE(hrn.IsNull());
+    }
+    {
+      SCOPED_TRACE("Valid Catalog HRN without realm");
+      HRN hrn("hrn:here:data:::hereos-internal-test-v2");
+      EXPECT_FALSE(hrn.IsNull());
+    }
+    {
+      SCOPED_TRACE("Invalid Catalog HRN (missing catalog name)");
+      HRN hrn("hrn:here:data::test:");
+      EXPECT_TRUE(hrn.IsNull());
+    }
+  }
+  {
+    SCOPED_TRACE("Schema HRN Tests");
```

Comment:
Separate into new test case.
Probably you can do a separation based on ServiceType.